server/modules/analysis/manager/processingChainManager.js

```diff
@@ -1,14 +1,28 @@
 const db = require('../../../db/db')
+
 const ActivityLog = require('../../activityLog/activityLogger')
+
+const ProcessingChain = require('../../../../common/analysis/processingChain')
+
 const ProcessingChainRepository = require('../repository/processingChainRepository')
 
+const createChain = async (user, surveyId, cycle, client = db) =>
+  await client.tx(async t => {
+    const processingChain = await ProcessingChainRepository.insertChain(surveyId, cycle, t)
+    await ActivityLog.log(user, surveyId, ActivityLog.type.processingChainInsert, { processingChain }, t)
```

Comment:
I think the repository can return the uuid directly and save in the history the uuid . is there a reason to save all the processing chai in the activity log?

Reply:
if you want to recreate the same processing chain (with cycle, props, date created/modified etc) you have to store it entirely